### PR TITLE
Disable lint suggestions on my PRs

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #138054

The suggestions unusably clog up early draft PRs that are not necessarily lint clean yet. Making matters worse, even if I fix them I have to manually click through hundreds of comments to "Resolve" them even though I've fixed it. Disabling it on ghstack helps, but I occasionally do standard PRs via fbcode export mechanism. Opt me out.

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

### CODE REVIEW

**Review by huydhn — APPROVED** [review]
Just FYI, there is an issue to improve the UX here https://github.com/pytorch/test-infra/issues/5760.  But, till it's done, opt-out it is then.

**Review by malfet — APPROVED** [review]
image

File: .github/workflows/lint-autoformat.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
       contents: read
       pull-requests: write
     runs-on: lf.linux.2xlarge
-    if: ${{ github.repository_owner == 'pytorch' }}
+    if: ${{ github.repository_owner == 'pytorch' && github.event.pull_request.user.login != 'ezyang' }}
```
Comment (malfet) [inline .github/workflows/lint-autoformat.yml:15]: Please... 
```suggestion
    if: ${{ github.repository_owner == 'pytorch' && github.event.pull_request.user.login != 'ezyang' && github.event.pull_request.user.login != 'malfet' }}
```

Comment (Skylion007) [reply .github/workflows/lint-autoformat.yml:15]: Can we just not run them on draft PRs?

Comment (malfet) [reply .github/workflows/lint-autoformat.yml:15]: May be the right solution is just to disable the workflow or make it an opt-in. Or perhaps run it only if your workflows needs approval...

Comment (ezyang) [reply .github/workflows/lint-autoformat.yml:15]: sure

Comment (ezyang) [reply .github/workflows/lint-autoformat.yml]: a sensible opt-in strategy is have the lint error ci message say "to suggest these as changes on your PR, comment BLAH" and then the flow triggers on that

**Review by PaliC — APPROVED** [review]

**Pull request closed without merging**